### PR TITLE
SDL Core implementation of feature [0257] New vehicle data HandsOffSteering

### LINKED ISSUE #3126: [SDL 0257] New vehicle data HandsOffSteering
Proposal: [New vehicle data HandsOffSteering](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0257-New-vehicle-data-HandsOffSteering.md)

> This proposal is to add new vehicle data `HandsOffSteering` to know whether or not a driver has his or her hands on the steering wheel.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/857

### PR DESCRIPTION
**Pull request by ychernysheva**
Implements/Fixes #3126 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
Unit testing + ATF testing

### Summary
This PR contains sdl_core implementation of [SDL-0257] New vehicle data HandsOffSteering.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/appMain/sdl_preloaded_pt.json [hunk 1]
```diff
@@ -587,7 +587,8 @@
                             "odometer",
                             "tirePressure",
                             "vin",
-                            "wiperStatus"
+                            "wiperStatus",
+                            "handsOffSteering"
```
Comment (LitvinenkoIra) [inline src/appMain/sdl_preloaded_pt.json:591]: @LitvinenkoIra to move it to the "DrivingCharacteristics-3" group

**Comment by IGapchuk** [comment]
NOTE: Please be informed, that the commit 37076da6b0838d0280138528dc65a630d63ac8ae is temporary and should be removed before merge.

File: src/appMain/sdl_preloaded_pt.json [hunk 2]
```diff
@@ -4082,6 +4086,13 @@
                     "key": "OEM_REF_MY_KEY",
                     "type": "Struct",
                     "mandatory": false
+                },
+                {
+                    "name": "handsOffSteering",
```
Comment (IGapchuk) [inline src/appMain/sdl_preloaded_pt.json:4091]: @IGapchuk move it to the separate commit with comment that it's redundent

File: src/appMain/sdl_preloaded_pt.json (hunk 1 above)
Comment (IGapchuk) [reply src/appMain/sdl_preloaded_pt.json]: Done in the dd77569bc33db24ce371f3388bf730a5a197e684

File: src/appMain/sdl_preloaded_pt.json (hunk 2 above)
Comment (IGapchuk) [reply src/appMain/sdl_preloaded_pt.json:4092]: Done in the 52282cfeb092e92fe9972ea3a205ba69beac5b51

File: src/appMain/sdl_preloaded_pt.json [hunk 3]
```diff
@@ -660,7 +664,8 @@
                             "cloudAppVehicleID",
                             "odometer",
                             "tirePressure",
-                            "wiperStatus"
+                            "wiperStatus",
+                            "handsOffSteering"
```
Comment (IGapchuk) [inline src/appMain/sdl_preloaded_pt.json:668]: @IGapchuk remove it

Comment (IGapchuk) [reply src/appMain/sdl_preloaded_pt.json]: Done in 01f16e0ece8b2ad2ea6c86d800c05d6c0deed38a

**Comment by LitvinenkoIra** [comment]
@atiwari9 please review this PR

**Comment by atiwari9** [comment]
PR Approved.

Validated with ATF scripts [a58bbb8](https://github.com/smartdevicelink/sdl_atf_test_scripts/commit/a58bbb882f20b1aa8fa27cf94774f1c667708383) , test results: [Test_Report_HandsFreeSteering_a58bbb8.zip](https://github.com/smartdevicelink/sdl_atf_test_scripts/files/4788528/Test_Report_HandsFreeSteering_a58bbb8.zip)

**Comment by LitvinenkoIra** [comment]
@theresalech PR is ready for Livio review. Thank you!

File: src/appMain/sdl_preloaded_pt.json [hunk 4]
```diff
@@ -4082,6 +4086,13 @@
                     "key": "OEM_REF_MY_KEY",
                     "type": "Struct",
                     "mandatory": false
+                },
+                {
+                    "name": "handsOffSteering",
+                    "key": "OEM_REF_HANDS_OFF_STRNG",
+                    "type": "Boolean",
+                    "mandatory": false,
+                    "since": "6.2"
```
Comment (jacobkeeler) [inline src/appMain/sdl_preloaded_pt.json:4095]:
```suggestion
                    "since": "6.1"
```

**Comment by YarikMamykin** [comment]
> Should probably update the policy test `json` files as well:
> 
> 1. `src/components/policy/policy_external/test/json/sdl_preloaded_pt_for_merge_initial.json`
> 2. `src/components/policy/policy_external/test/json/sdl_preloaded_pt_for_merge_latest.json`
> 3. `src/components/policy/policy_regular/test/json/sdl_preloaded_pt_for_merge_initial.json`
> 4. `src/components/policy/policy_regular/test/json/sdl_preloaded_pt_for_merge_latest.json`

@jacobkeeler files updated in https://github.com/smartdevicelink/sdl_core/pull/3394/commits/d2b04e09e77798bda1ac9a83fa1edef795e460b8

File: .gitmodules [hunk 5]
```diff
@@ -3,4 +3,5 @@
 	url = https://github.com/smartdevicelink/jsoncpp.git
 [submodule "tools/rpc_spec"]
 	path = tools/rpc_spec
-	url = https://github.com/smartdevicelink/rpc_spec.git
+	url = https://github.com/LuxoftSDL/rpc_spec.git
+	branch = feature/0257_vehicle_data_handsOffSteering
```
Comment (jacobkeeler) [inline .gitmodules:7]:
```suggestion
	url = https://github.com/smartdevicelink/rpc_spec.git
```
Reminder to revert this bit before merge

Comment (AKalinich-Luxoft) [reply .gitmodules]: @jacobkeeler reverted in https://github.com/smartdevicelink/sdl_core/pull/3394/commits/9fe091622f5919005c4961995a8a4f6960008c52

**Comment by AKalinich-Luxoft** [comment]
> Note: will need to revert the `rpc_spec` submodule changes before merging

@jacobkeeler conflicts have been resolved and PR is ready for merge
